R/readGWLdata.R

```diff
@@ -10,50 +10,142 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and limitations under the License.
 
-#' Read in groundwater data from file downloaded from GWL tool
-#'
-#' Read in groundwater data from file downloaded from GWL tool
-#' @param path The path to the csv
-#' @param emsID The EMS ID of the well
-#' @export
+
+
+
+#' Retrive and format groundwater data from BC Government GWL site
+#' 
+#' Go to <http://www.env.gov.bc.ca/wsd/data_searches/obswell/map/> to find your 
+#' well of interest.
+#' 
+#' Note that well water levels are measured in meters below the ground. Thus 
+#' higher values represent deeper levels. Therefore 
+#' \code{Historical_Daily_Minimum} values will be greater than 
+#' \code{Historical_Daily_Maximum} values, as they represent a lower water 
+#' level.
+#' 
+#' Daily averages (\code{which = 'daily'}) are pre-calculated. Note that they 
+#' only cover days marked as "Validated" in the full hourly dataset.
+#' 
+#' @param well Character. The well number, accepts either \code{OW000} or 
+#'   \code{000} format
+#' @param which Character. Which data to retrieve \code{all} hourly data, 
+#'   \code{recent} hourly data, or all \code{daily} averages.
+#' @param url Character. Override the url location of the data.
+#' @param quiet Logical. Suppress progress messages?
+#' 
 #' @return A dataframe of the groundwater level observations
+#' 
 #' @examples \dontrun{
-#'
+#' 
+#' all_309 <- get_gwl(well = 309)
+#' recent_309 <- get_gwl(well = "OW309", which = "recent")
+#' daily_avg_309 <- get_gwl(well = "OW309", which = "daily")
 #'}
-readGWLdata <- function(path, emsID = NULL) {
+#'
+#' @export
+get_gwl <- function(well, which = "all", url = NULL, quiet = FALSE) {
   
-  if (!inherits(path, "textConnection") && !file.exists(path)) {
-    stop(paste0("The file ", path, "does not exist."))
-  }
+  if(!(which %in% c("all", "recent", "daily"))) stop("type must be either 'all', 'recent', or 'daily'.")
```

Comment:
What do you think about using `which = c("all", "recent", "daily")` in the argument list, and then `which <- match.arg(which)` to check?

Reply:
I have to admit it never occurred to me, but yes, that would totally work. The only small thing I don't like is that the error message is: 
```
Error in match.arg(which) : 
  'arg' should be one of “all”, “recent”, “daily”
```
Which, to an R newbie, isn't really a clear error message.

But either way, 'type' should be 'which' in the original!

Reply:
I hope you don't mind, but this ispired me to write a function `arg_match` that does a better job of this. I implemented it for `get_gwl` in https://github.com/bcgov/bcgroundwater/pull/10/commits/65acc501741a1d24359be7473ee7f8445ae45c9d

Reply:
I definitely don't mind, that's a great function! I like it much better than the original.

Reply:
I think I'll make use of it in a lot of other places!